src/vma/sock/sockinfo_udp.cpp

```diff
@@ -1371,7 +1371,7 @@ void sockinfo_udp::handle_recv_timestamping(struct cmsg_state *cm_state)
 	}
 
 	if (m_n_tsing_flags & SOF_TIMESTAMPING_RAW_HARDWARE) {
-		tsing.hwtimeraw = packet->path.rx.hw_timestamp;
+		packet->path.rx.p_ib_ctx_time_converter->convert_hw_time_to_system_time(packet->path.rx.hw_raw_timestamp, &tsing.hwtimeraw);
```

Comment:
_SOF_TIMESTAMPING_SOFTWARE_ is for sw timestamp. i cant understand why we should move the conversion to there...
There is no need to keep the value - we are coping the converted value into the user buffer :

```
void sockinfo_udp::handle_recv_timestamping(struct cmsg_state *cm_state)
{
       .....

	if (m_n_tsing_flags & SOF_TIMESTAMPING_SOFTWARE) {
		tsing.systime = packet->path.rx.sw_timestamp;
	}

	if (m_n_tsing_flags & SOF_TIMESTAMPING_RAW_HARDWARE) {
		packet->path.rx.p_ib_ctx_time_converter->convert_hw_time_to_system_time(packet->path.rx.hw_raw_timestamp, &tsing.hwtimeraw);
	}

	insert_cmsg(cm_state, SOL_SOCKET, SO_TIMESTAMPING, &tsing, sizeof(tsing));
}

```